src/components/Calendar/Calendar.tsx

```diff
@@ -190,6 +192,29 @@ const makeAllDayEventsInclusive = (endDateExclusiveEvents: CalendarEvent[]): Cal
     });
 };
 
+const initialLocations = [
+    "Vauxhall Hope Church",
+    "Waterloo - St George the Martyr",
+    "Waterloo - Oasis",
+    "Waterloo - St Johns",
+    "Brixton Hill - Methodist Church",
+    "N&B - Emmanuel Church",
+    "Streatham - Immanuel and St Andrew",
+];
+
+const filterEventsByLocation = (
+    locations: string[],
+    allEvents: CalendarEvent[]
+): CalendarEvent[] => {
+    const viewedEvents: CalendarEvent[] = [];
+    allEvents.forEach((event: CalendarEvent) => {
+        if (locations.includes(event.location)) {
+            viewedEvents.push(event);
+        }
+    });
+    return makeAllDayEventsInclusive(viewedEvents);
+};
```

Comment:
can this just be smth like:

` return makeAllDayEventsInclusive(allEvents.filter(event => locations.includes(event.location))`

Reply:
Think having it in two separate lines is slightly cleaner, but you are correct!